File: 2020/Nim/day3.nim
import aoc
import strutils

func solve*(input:string): (int, int) =
    let data = input.splitLines
    func trees(dx, dy: int = 1): int =
        for y in countup(0, data.len-1, dy):
            result += int(data[y][dx * y mod data[0].len] == '#')
    let part1 = trees(3)
    let part2 = trees(1)*trees(3)*trees(5)*trees(7)*trees(1, 3)
    return (part1, part2)
                

File: 2020/Python/day3.py
from aoc import *

def solve(inp):
    inp = inp.splitlines()
    n = len(inp[0])
    def slope(q,skip=1):
        res = 0
        for i,row in enumerate(inp[::skip]):
            res += row[q*i % n] == "#"
        return res
    part1 = slope(3)
    part2 = slope(1)*slope(3)*slope(5)*slope(7)*slope(1,2)    
    return part1, part2


run(solve)